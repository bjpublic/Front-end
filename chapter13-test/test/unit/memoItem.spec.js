import MemoItem from "../../src/memoItem";

describe("MemoItem", () => {
  let memoItem, el;

  beforeEach(() => {
    el = document.createElement("div");
    const content = "item content";

    memoItem = new MemoItem(el, { content });
  });

  it("memoItem 요소가 렌더링된다.", () => {
    expect(el).toContainElement(memoItem.itemEl);
  });

  it("memoItem 요소는 스냅샷과 일치해야 한다.", () => {
    expect(memoItem.itemEl).toMatchSnapshot();
  });

  it("destroy() 메서드 호출 시 삭제된다.", () => {
    memoItem.destroy();

    expect(el).not.toContainElement(memoItem.itemEl);
  });
});
