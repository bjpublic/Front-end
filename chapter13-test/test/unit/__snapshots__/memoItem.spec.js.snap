// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoItem memoItem 요소는 스냅샷과 일치해야 한다. 1`] = `
<li
  class="list-group-item d-flex justify-content-between align-items-center"
>
  <span
    class="text-break"
  >
    item content
  </span>
  <button
    class="btn btn-danger btn-sm"
    data-item-id="1"
  >
    X
  </button>
</li>
`;
